# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.h
# Commit: 5737845e8af2
# Full Hash: 5737845e8af228fab6e5e622863663faa50f93e8
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
# ==============================================================================

diff -r 5dfb9fc15fba -r 5737845e8af2 js/src/wasm/WasmTypes.h
--- a/js/src/wasm/WasmTypes.h	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/wasm/WasmTypes.h	Fri Oct 23 18:17:38 2020 +0000
@@ -3267,7 +3267,6 @@
 
   uint8_t* rawCaller() const { return callerFP_; }
   TlsData* tls() const { return tls_; }
-  Instance* instance() const { return tls()->instance; }
 
   Frame* wasmCaller() const {
     MOZ_ASSERT(!callerIsExitOrJitEntryFP());
@@ -3427,7 +3426,7 @@
   static DebugFrame* from(Frame* fp);
   Frame& frame() { return frame_; }
   uint32_t funcIndex() const { return funcIndex_; }
-  Instance* instance() const { return frame_.instance(); }
+  Instance* instance() const;
   GlobalObject* global() const;
   bool hasGlobal(const GlobalObject* global) const;
   JSObject* environmentChain() const;
